viewing_party/party.py

```diff
@@ -1,23 +1,139 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        return {
+        "title": title,
+        "genre": genre, 
+        "rating": rating
+        }
+    else:
+        return None
+        
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data 
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # watchlist = user_data["watchlist"]
+    # watched = user_data["watched"]
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    list_of_ratings = []
+    for movie in user_data["watched"]:
+        list_of_ratings.append(movie["rating"])
+    if len(user_data["watched"]) == 0:
+        avg_rating = 0
+    else:
+        avg_rating = sum(list_of_ratings)/len(list_of_ratings)
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+    for movie in user_data["watched"]:
+        genre_list.append(movie["genre"])
+
+    if user_data["watched"] == []:
+        return None
+    else:
+        counter = 0
+        for genre in genre_list:
+            frequency = genre_list.count(genre)
+            if frequency > counter:
+                counter = frequency
+                most_watched_genre = genre
+            return most_watched_genre
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_watched = []
+    friends_movies = set()
+    users_movies = set()
+    # We want to access "friends"
+    # and for every friend in there, we want to 
+    # access a movie in their "watched" list of
+    # movie dictionaries.
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_movies.add(movie["title"])
+    for movie in user_data["watched"]:
+        users_movies.add(movie["title"])
+    unique_movies = users_movies - friends_movies
+    for movie in user_data["watched"]:
+        if movie["title"] in unique_movies:

```

Comment:
👍 Here, since you have a set of the unique titles, this is a nice constant time lookup.